subsys/net/lib/zperf/zperf_udp_uploader.c

```diff
@@ -268,7 +316,7 @@ static int udp_upload(int sock, int port,
 		if (IS_ENABLED(CONFIG_NET_ZPERF_LOG_LEVEL_DBG)) {
 			if (print_time >= loop_time) {
 				NET_DBG("nb_packets=%u\tdelay=%u\tadjust=%d",
-					nb_packets, (unsigned int)delay,
+					nb_packets, (unsigned int)compensate_delay,
```

Comment:
I don't think there's a need for yet another `compensate_delay` variable, just update the `delay` value if needed. There'd be no need for the `else` branches above then.